CRM/Contact/Form/Search/Custom/Proximity.php

```diff
@@ -39,21 +36,6 @@ public function __construct(&$formValues) {
     unset($this->_formValues['uf_group_id']);
     unset($this->_formValues['component_mode']);
     unset($this->_formValues['operator']);
-
-    if (!empty($this->_formValues)) {
-      // add the country and state
-      self::addGeocodingData($this->_formValues);
-      $this->_latitude = $this->_formValues['geo_code_1'];
-      $this->_longitude = $this->_formValues['geo_code_2'];
-
-      if ($this->_formValues['prox_distance_unit'] == "miles") {
```

Comment:
won't this still be needed @monishdeb ?

Reply:
@seamuslee001 nope. Because, CRM_Contact_BAO_ProximityQuery::process($this, $formValues); already handles the conversion [here](https://github.com/civicrm/civicrm-core/blob/master/CRM/Contact/BAO/ProximityQuery.php#L308). So here's the workflow: 
1. at L201 below we are constructing the search SQL and in the param we are passing prox_distance_unit and prox_distance values at L181. 
2. Eventuall it calls [CRM_Contact_BAO_ProximityQuery::process($this, $formValues)](https://github.com/civicrm/civicrm-core/blob/master/CRM/Contact/BAO/Query.php#L2012)  
3. And then the distance is converted to meters on basis of distance unit at [here](https://github.com/civicrm/civicrm-core/blob/master/CRM/Contact/BAO/ProximityQuery.php#L308)